### PR TITLE
Remove sphinx static and template dir

### PR DESCRIPTION
**Pull request by stsewd**
This is already handled by the rtd-sphinx-extension.

We don't override the template from rtd (#1946)

Half solves #5029

### CODE REVIEW

File: readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl [hunk 1]
```diff
@@ -37,7 +37,6 @@ if not 'html_static_path' in globals():
     html_static_path = []
 if os.path.exists('_static'):
     html_static_path.append('_static')
```
Comment (stsewd) [inline readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl]: I'm tempted to remove this whole block too

**Comment by stsewd** [comment]
Tested it locally, no more warnings, everything is working.

**Comment by stsewd** [comment]
> this is not something needed for the corporate site

I didn't find any reference in .com

> we don't need to override any Sphinx template never again :)
by removing this, we still do have another way to override a Sphinx template (using our extension for example)

I think we should stay away of modifying the theme inside the rtd, all that should be managed by the extension, that way is less likely to break in case we update sphinx, or users can pin to a compatible version of the extension.

**Comment by humitos** [comment]
OK! I suppose we are fine, then.

**Pull request merged**